megamek/src/megamek/utilities/ai/DatasetParser.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2025 - The MegaMek Team. All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+package megamek.utilities.ai;
+
+import megamek.common.*;
+import megamek.common.util.StringUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Parses a dataset from one or more files and turns it into a training dataset.
+ * The dataset currently expected is the game_actions_log.tsv file generated by the megamek client hosting the game if the
+ * option is enabled
+ * @author Luana Coppio
+ */
+public class DatasetParser {
+
+    // attackAction is not being considered right now
+    private final List<ActionAndState> actionAndStates = new ArrayList<>();
+    private final Map<Integer, Entity> entities = new HashMap<>();
+    private int idOffset = 0;
+    private int highestEntityId = 0;
+
+    /**
+     * Parses a dataset from a file. Can be chained with other parse calls to create a large single training dataset.
+     * @param file The file to parse
+     * @return The parser instance
+     */
+    public DatasetParser parse(File file) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.startsWith("PLAYER_ID\tENTITY_ID\tCHASSIS\tMODEL\tFACING\tFROM_X\tFROM_Y\tTO_X\tTO_Y\tHEXES_MOVED\tDISTANCE\tMP_USED\tMAX_MP\tMP_P\tHEAT_P\tARMOR_P\tINTERNAL_P\tJUMPING\tPRONE\tLEGAL\tSTEPS")) {
+                    // Parse action line
```

Comment:
These need to be an enum or something.

Reply:
I see no benefit into making this an enum, care to elaborate?

Reply:
Oh, I see the issue: I was thinking this was checking for matches from a set, not for a single very long line with tab separation.
At any rate, this should definitely be a constant (something like "ACTION_LINE_HEADER") for clarity and ease of editing later.